src/DynamicData/Kernel/IndexMinusOneException.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+namespace DynamicData.Kernel;
+
+/// <inheritdoc />
+public class IndexMinusOneException : Exception
+{
+    private const string DefaultMessage = "Index of minus one detected. Please raise an issue for DynamicData with an exact re-production";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="IndexMinusOneException"/> class.
+    /// </summary>
+    /// <param name="context"> Some context.</param>
+    /// <param name="member"> The calling member.</param>
+    /// <param name="line"> The calling line number..</param>
+    public IndexMinusOneException(string context, [CallerMemberName] string? member = null, [CallerLineNumber] int line = 0)
```

Comment:
We should throw this beat whenever we detect a -1 index (unless of course we can handle it)

Reply:
`MissingIndexException`? `UnspecifiedIndexException`?

Reply:
Now uses UnspecifiedIndexException